src/component/parser/json/converter/CollaborationConverter.ts

```diff
@@ -18,31 +18,13 @@ import { MessageFlow } from '../../../../model/bpmn/internal/edge/Flow';
 import { TCollaboration } from '../../../../model/bpmn/json/baseElement/rootElement/collaboration';
 import { TParticipant } from '../../../../model/bpmn/json/baseElement/participant';
 import { TMessageFlow } from '../../../../model/bpmn/json/baseElement/baseElement';
-import { ensureIsArray } from './ConverterUtil';
-
-const convertedParticipantsById: Map<string, Participant> = new Map();
-const convertedParticipantsByProcessRef: Map<string, Participant> = new Map();
-const convertedMessageFlows: Map<string, MessageFlow> = new Map();
-
-export function findProcessRefParticipant(id: string): Participant {
-  return convertedParticipantsById.get(id);
-}
-
-export function findProcessRefParticipantByProcessRef(processRef: string): Participant {
-  return convertedParticipantsByProcessRef.get(processRef);
-}
-
-export function findMessageFlow(id: string): MessageFlow {
-  return convertedMessageFlows.get(id);
-}
+import { ConvertedElements, ensureIsArray } from './utils';
 
 export default class CollaborationConverter {
+  constructor(readonly convertedElements: ConvertedElements) {}
+
   deserialize(collaborations: string | TCollaboration | (string | TCollaboration)[]): void {
     try {
-      convertedParticipantsById.clear();
```

Comment:
We have to clean the maps before each new parsing, because it's keep the old converted elements of the previous file.
Some elements can be in conflict if there are the same id.

Reply:
Forgot my comment. I didn't see that we instantiate a new BPMNParser (and new ConvertedElements) at each loading ^^